test/LocatedTriplesTest.cpp

```diff
@@ -0,0 +1,704 @@
+      numTriplesPerBlock,
+      [](auto p) -> testing::Matcher<const LocatedTriplesPerBlock&> {
+        auto [blockIndex, insertsAndDeletes] = p;
+        return numTriplesBlockwise(blockIndex, insertsAndDeletes);
+      });
+  return testing::AllOf(numBlocks(blocks), numTriplesTotal(triples),
+                        testing::AllOfArray(blockMatchers));
+};
+}  // namespace Matchers
+namespace m = Matchers;
+
+// Fixture with helper functions.
+class LocatedTriplesTest : public ::testing::Test {
+ protected:
+  // Make `LocatedTriplesPerBlock` from a list of `LocatedTriple` objects (the
+  // order in which the objects are given does not matter).
+  LocatedTriplesPerBlock makeLocatedTriplesPerBlock(
+      const std::vector<LocatedTriple>& locatedTriples) {
+    LocatedTriplesPerBlock result;
+    result.add(locatedTriples);
+    return result;
+  }
+};
+
+auto IT = [](const auto& c1, const auto& c2, const auto& c3) {
+  return IdTriple{V(c1), V(c2), V(c3)};
+};
+
+// Test the method that counts the number of `LocatedTriple's in a block.
+TEST_F(LocatedTriplesTest, numTriplesInBlock) {
+  using LT = LocatedTriple;
+  // Set up lists of located triples for three blocks.
+  auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock(
+      {LT{1, IT(10, 1, 0), false}, LT{1, IT(10, 2, 1), false},
+       LT{1, IT(11, 3, 0), true}, LT{2, IT(20, 4, 0), true},
+       LT{2, IT(21, 5, 0), true}, LT{4, IT(30, 6, 0), true},
+       LT{4, IT(32, 7, 0), false}});
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(3, 7, {{1, {1, 2}}, {2, {2, 0}}, {3, {0, 0}}, {4, {1, 1}}}));
+
+  auto handles = locatedTriplesPerBlock.add(
+      {LT{3, IT(25, 5, 0), true}, LT{4, IT(31, 6, 1), false}});
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(4, 9, {{1, {1, 2}}, {2, {2, 0}}, {3, {1, 0}}, {4, {1, 2}}}));
+
+  locatedTriplesPerBlock.erase(3, handles[0]);
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(3, 8, {{1, {1, 2}}, {2, {2, 0}}, {3, {0, 0}}, {4, {1, 2}}}));
+
+  locatedTriplesPerBlock.erase(4, handles[1]);
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(3, 7, {{1, {1, 2}}, {2, {2, 0}}, {3, {0, 0}}, {4, {1, 1}}}));
+
+  locatedTriplesPerBlock.clear();
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(0, 0, {{1, {0, 0}}, {2, {0, 0}}, {3, {0, 0}}, {4, {0, 0}}}));
+}
+
+TEST_F(LocatedTriplesTest, erase) {
+  using LT = LocatedTriple;
+  // Set up lists of located triples for three blocks.
+  auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock(
+      {LT{1, IT(10, 1, 0), false}, LT{1, IT(10, 2, 1), false},
+       LT{1, IT(11, 3, 0), true}, LT{2, IT(20, 4, 0), true},
+       LT{2, IT(21, 5, 0), true}, LT{4, IT(30, 6, 0), true},
+       LT{4, IT(32, 7, 0), false}});
+
+  auto handles = locatedTriplesPerBlock.add({LT{1, IT(15, 15, 15), false}});
+  EXPECT_THROW(locatedTriplesPerBlock.erase(5, handles[0]),
+               ad_utility::Exception);
```

Comment:
Maybe you can merge the tests for insert and erase, and also test the contents.

Reply:
What do you mean by content? The map that contains the located triples is private.

Reply:
Okay, 
So either you expose a const& getter for testing (not that much speaks against it), 
Or you do something like a true end to end, where you merge this with an empty block, and then you can check the contents. 
But the first one is probably cleaner, as it mixes less functions. But for the test, if a add the triple {15, 15, 15} , I want to assert that it is contained afterwards.

Reply:
The second would require a block that only contains the triples to be deleted and must then only contains those that were inserted. That would indeed be overly complicated. I have now done it with `FRIEND_TEST`.